crate/crash/command.py

```diff
@@ -618,8 +621,60 @@ def save_and_exit():
     save_and_exit()
 
 
+INFINITE_TIMEOUT = -1
+INFINITE_TIMEOUTS = [None, INFINITE_TIMEOUT, str(INFINITE_TIMEOUT)]
+
+
+def _decode_timeout(value):
+    """
+    Decode single timeout value, respecting the `INFINITE_TIMEOUT` surrogate.
+    """
+    if value in INFINITE_TIMEOUTS:
+        return None
+    else:
+        return float(value)
+
+
+def _decode_timeouts(timeout):
+    """
+    Decode connect/read timeout values from tuple or string.
+
+    Variant 1: (connect, read)
+    Variant 2: connect,read
+    """
+
+    if timeout is None:
+        timeouts = (INFINITE_TIMEOUT, INFINITE_TIMEOUT)
+    elif isinstance(timeout, (int, float)):
+        timeouts = (timeout, INFINITE_TIMEOUT)
+    elif isinstance(timeout, str):
+        timeouts = timeout.split(",")
+    elif isinstance(timeout, (list, tuple)):
+        timeouts = timeout
+    else:
+        raise TypeError(f"Decoding timeout values failed. Invalid type: {type(timeout)}")
+
+    if len(timeouts) == 1:
+        connect_timeout, read_timeout = timeouts[0], INFINITE_TIMEOUT
+    elif len(timeouts) == 2:
+        connect_timeout, read_timeout = timeouts[0], timeouts[1]
+    else:
+        raise ValueError(f"Decoding timeout values failed: {timeout}")
```

Comment:
```suggestion
        raise ValueError(f"Cannot decode timeout {timeout}, expected format `<connect_sec>,<read_sec>`")
```

Reply:
Thanks. I've adjust this error message, and also aligned the other one for the `TypeError` a few lines above.